lib/score.rb

```diff
@@ -0,0 +1,34 @@
+class Score
+  attr_reader :first_roll_pins, :second_roll_pins, :spare, :strike
+  attr_accessor :score
+  def initialize
+    @first_roll_pins
+    @second_roll_pins
+    @spare = false
+    @strike = false
+    @score
+  end
+
+  def first_roll(pins:)
+    @first_roll_pins = pins
+    determine_strike
+  end
+
+  def second_roll(pins:)
+    @second_roll_pins = pins
+    determine_spare
+  end
+
+  def determine_strike
+    @strike = true if first_roll_pins == 10
+  end
+
+  def determine_spare
+    @spare = true if @first_roll_pins + @second_roll_pins == 10 && @strike == false
+  end
+
+  def calculate_score
+    !@spare && !@strike ?
+    @score = @first_roll_pins + @second_roll_pins : @score = 0
```

Comment:
It looks like if it's a spare or strike then Score doesn't actually calculate the score? Is that handled in Game?

Reply:
Yep! As I wanted to add the bonus to the scoring from the previous frame, so game adds bonus to prior frame total (old scores are stored in an array in Game)